modules/personalize_taxonomy_context/plugins/visitor_context/TaxonomyContext.inc

```diff
@@ -0,0 +1,93 @@
+<?php
+
+/**
+ * @file
+ * Provides a visitor context plugin for taxonomy.
+
+ */
+
+class TaxonomyContext extends PersonalizeContextBase {
+  /**
+   * Implements PersonalizeContextInterface::create().
+   */
+  public static function create(PersonalizeAgentInterface $agent = NULL, $selected_context = array()) {
+    return new self($agent, $selected_context);
+  }
+
+  /**
+   * Implements PersonalizeContextInterface::getOptions().
+   */
+  public static function getOptions() {
+    $vocabularies = taxonomy_get_vocabularies();
+    $options = array();
+    foreach ($vocabularies as $vocabulary) {
+      $machine_name = $vocabulary->machine_name;
+      $options[$machine_name] = array(
+        'name' => $vocabulary->name,
+        'group' => 'Taxonomy',
+        'id' => $vocabulary->vid,
+      );
+    }
+    return $options;
+  }
+
+  /**
+   * Implements PersonalizeContextInterface::getPossibleValues().
+   */
+  public function getPossibleValues($limit = FALSE) {
+    $possible_values = array();
+    $options = $this->getOptions();
+    $free_tagging_vocabulary_machine_names = SELF::getFreeTaggingVocabulariesMachineNames();
```

Comment:
Unfortunately that documentation is slightly misleading. It's not the vocabulary that's free-tagging, it's the field widget being used when a term reference field is added to a content type. The same vocabulary can in theory be provided as free-tagging in one content type and as a select list in another. Also, it sounds like the real issue here isn't whether or not free-tagging is involved, but how many terms there are. There could be hundreds of terms in any vocabulary, regardless of whether it is used for a free-tagging term reference field. So if we're going to limit which vocabularies show up here, it should be limited on that basis, not on some attempt to guess whether they're being used as free-tagging vocabularies.
Also, regarding point 2 above, note that none of this affects how terms are sent to Lift Web - this only affects the ability to use a vocabulary for fixed targeting.